internal/services/network/application_gateway_resource.go

```diff
@@ -1019,6 +1019,18 @@ func resourceApplicationGateway() *pluginsdk.Resource {
 													Type:     pluginsdk.TypeString,
 													Optional: true,
 												},
+
+												"components": {
+													Type:     pluginsdk.TypeString,
+													Optional: true,
+													Computed: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														"path_only",
+														"query_string_only",
+														"",
+													}, true),
```

Comment:
since we're explicitly comparing the values this needs to be case-sensitive, and there's no need for this to be computed?

```suggestion
													ValidateFunc: validation.StringInSlice([]string{
														"path_only",
														"query_string_only",
													}, false),
```